daemon/daemon.go

```diff
@@ -82,21 +79,22 @@ func (d *Daemon) Main(serve func(string, string)) int {
 // client.
 func (d *Daemon) Spawn() error {
 	binPath := d.BinPath
-	// Determine binPath.
-	if binPath == "" {
-		if len(os.Args) > 0 && path.IsAbs(os.Args[0]) {
-			binPath = os.Args[0]
-		} else {
-			// Find elvish in PATH
-			paths := strings.Split(os.Getenv("PATH"), ":")
-			result, err := util.Search(paths, "elvish")
-			if err != nil {
-				return errors.New("cannot find elvish: " + err.Error())
-			}
-			binPath = result
+	if !filepath.IsAbs(binPath) {
```

Comment:
I feel that it is better to use the homegrown searching utility (`util.Search`) in place of `exec.LookPath` in consistency with the rest of Elvish.